drivers/azure/azureutil/securityrules.go

```diff
@@ -0,0 +1,319 @@
+				SourceAddressPrefix:      to.StringPtr("*"),
+				DestinationAddressPrefix: to.StringPtr("*"),
+				SourcePortRange:          to.StringPtr("*"),
+				DestinationPortRange:     to.StringPtr(openPort.Port),
+				Access:                   network.SecurityRuleAccessAllow,
+				Direction:                network.SecurityRuleDirectionInbound,
+				Protocol:                 openPort.Proto,
+			},
+		}
+	}
+	// Log open ports to be deleted and added
+	if len(openPortsToBeDeleted) > 0 || len(openPortsToBeAdded) > 0 {
+		removedPorts := make([]string, len(openPortsToBeDeleted))
+		addedPorts := make([]string, len(openPortsToBeAdded))
+		i := 0
+		for openPort := range openPortsToBeDeleted {
+			removedPorts[i] = openPort
+		}
+		i = 0
+		for openPort := range openPortsToBeAdded {
+			addedPorts[i] = openPort
+		}
+		log.Info("Updating security rule configuration", logutil.Fields{
+			"openPortsRemoved": removedPorts,
+			"openPortsAdded":   addedPorts,
+		})
+	}
+	// Remove all unnecessary open ports and release the priorities
+	for _, rule := range openPortsToBeDeleted {
+		delete(takenPriorities, to.Int32(rule.Priority))
+		rules = removeSecurityRule(rules, rule)
+	}
+	// Add any necessary open ports
+	newRules := *rules
+	priority := int32(1000)
+	for _, rule := range openPortsToBeAdded {
+		// Find a priority that can be given to this rule
+		_, taken := takenPriorities[priority]
+		for taken {
+			priority++
+			_, taken = takenPriorities[priority]
+		}
+		rule.Priority = to.Int32Ptr(priority)
+		priority++
+		newRules = append(newRules, rule)
+	}
+	return &newRules, nil
+}
+
+// checkIfDockerConflict does a rudimentary check to see if the rule could potentially conflict with the ability
+// for the docker port to be accessible. If the rule does not pass all rudimentary checks, it returns an error.
+func checkIfDockerConflict(rule network.SecurityRule, dockerPort int) error {
+	f := logutil.Fields{
+		"name":        to.String(rule.Name),
+		"description": to.String(rule.Description),
+		"priority":    to.Int32(rule.Priority),
+		"access":      rule.Access,
+		"direction":   rule.Direction,
+		"protocol":    rule.Protocol,
+		"portRanges":  getDestinationPorts(rule),
+	}
+	priority := to.Int32(rule.Priority)
+	if priority > dockerPriority {
+		return nil
+	}
+	purpose := strings.ToLower(fmt.Sprintf("%s %s %s", rule.Access, rule.Direction, rule.Protocol))
+	if purpose == "deny inbound tcp" || purpose == "deny inbound *" {
+		// check if docker port is covered by rule
+		for _, pr := range getDestinationPorts(rule) {
+			portWithinRange, err := checkIfPortInPortRange(dockerPort, pr)
+			if err != nil {
+				return fmt.Errorf("%s: %s", err, f)
+			}
+			if portWithinRange {
+				return fmt.Errorf("rule %s set to %s requests to the Docker port %d: %s",
+					to.String(rule.Name), purpose, dockerPort, f)
+			}
+		}
+	}
+	return nil
+}
+
+func checkIfPortInPortRange(port int, portRange string) (bool, error) {
```

Comment:
seems pointless to have this return an error from this, nothing bubbles out except the one you write, i feel you should just make this return a bool and do the check in the caller

Reply:
agreed, changing the behavior of `checkIfPortInPortRange` to return false if the port is not within the port range or if the port range is malformed. In the latter case, we can't identify conflicts in a malformed rule anyways so we ignore it and move on.